janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/thrift/thriftpool/CTConnectionFactory.java

```diff
@@ -135,8 +143,15 @@ public boolean validateObject(String key, CTConnection c) {
                           c, curCfg, c.getConfig());
             }
         }
+        
+        boolean isOpen = c.isOpen();
+        try {
+            c.getClient().describe_version();
```

Comment:
Ah, Ok, understood. If the connection is used only for validation after it's been used already then this is really is the only way.